File: 2033/2033.java
class Solution {
    public int minOperations(int[][] grid, int x) {
        int result = 0;
        //create a
        int n = grid.length * grid[0].length;
        int[] a = new int[n];
        int index = 0;
        for (int[] row : grid) {
            for (int num : row) {
                a[index++] = num;
            }
        }
        Arrays.sort(a);
        int common = a[Math.floorDiv(n,2)];
        int commonX = common % x;
        for (int num : a){
            if (num%x != commonX) return -1;
            result += Math.floorDiv(Math.abs(common-num), x);
        }


        return result;
    }
}

File: 2033/2033.ts
function minOperations(grid: number[][], x: number): number {
    let result: number = 0;
    const a = grid.flat();
    a.sort((a, b) => a - b); //sort a as numbers instead of default str
    const n = a.length;
    const common: number = a[Math.floor(n/2)];
    const commonX: number = common%x;
    for (const num of a) {
        if (num%x != commonX) return -1
        result += Math.floor(Math.abs(common-num) / x)
    }
    return result;
};
